packages/frontend/src/components/DebouncedSearchInput/index.tsx

```diff
@@ -13,31 +18,54 @@ export default function DebouncedSearchInput({
   onChange,
   searchValue,
 }: DebouncedSearchInputProps): React.ReactElement {
+  const [tempSearchValue, setTempSearchValue] = useState(searchValue)
+
   const handleSearchInputChangeDebounced = useMemo(
     () => debounce(onChange, 500),
     [onChange],
   )
 
   const onSearchInputChange = useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
+      setTempSearchValue(event.target.value)
       handleSearchInputChangeDebounced(event.target.value)
     },
     [handleSearchInputChangeDebounced],
   )
 
+  const clearSearch = useCallback(() => {
+    setTempSearchValue('')
+    onChange('')
+    handleSearchInputChangeDebounced.cancel()
+  }, [onChange, handleSearchInputChangeDebounced])
+
   return (
     <InputGroup>
       <InputLeftElement h="100%">
-        <Icon as={BiSearch} style={{ height: 20, width: 20 }} />
+        <Icon as={BiSearch} h={5} w={5} />
       </InputLeftElement>
       <Input
         type="text"
-        w="full"
+        w={{ base: '100%', md: '20rem' }}
+        maxW="100%"
         fontSize="md"
         onChange={onSearchInputChange}
         defaultValue={searchValue}
+        value={tempSearchValue}
         placeholder="Search"
       />
+      <InputRightElement h="100%">
+        <Icon
+          as={BiSolidXCircle}
+          cursor="pointer"
+          opacity={0.6}
+          _hover={{ opacity: 1 }}
+          h={4}
+          w={4}
+          display={tempSearchValue ? 'block' : 'none'}
+          onClick={clearSearch}
+        />
+      </InputRightElement>
```

Comment:
```suggestion
      {tempSearchValue && (
        <InputRightElement h="100%">
          <Icon
            as={BiSolidXCircle}
            cursor="pointer"
            opacity={0.6}
            _hover={{ opacity: 1 }}
            onClick={clearSearch}
          />
        </InputRightElement>
      )}
```
cannot click on the right side when `tempSearchValue` is empty and `boxSize` is default 16px alr